CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+
+    1. Packages should loaded at the top of the script.
+    2. The easiest way to include data is to use `dput()` to generate the R code to recreate it. But even better is if you can create a data.frame() with just a handful of rows and columns that still illustrates the problem.
+    3. Spend a little bit of time ensuring that your code is easy for others to read: (1) Make sure you've used spaces and your variable names are concise, but informative. (2) Use comments to indicate where your problem lies. (3) Do your best to remove everything that isn't related to the problem. The shorter your code is, the easier it is to understand.
+<br><br>
+
+* **Improve documentation.** Noticed a typo? Think a function could use a better example? Improvements to the documentation is very welcome!
+
+* **Contribute code.** If you'd like to fix a bug or implement a new feature, have a look at the issue list and leave a comment on things you want to work on. Also see the development guidelines below.
+
+
+# Development guidelines
+
+We follow the [Gitflow](https://nvie.com/posts/a-successful-git-branching-model/) development model and the guidelines from [R packages](https://r-pkgs.org/) by Hadley Wickham and Jenny Bryan.
+
+1. Create a branch in git and make your changes. 
+
+    * Write your code.
+    * Restart R Session Cmd+Shift+F10 (Ctrl+Shift+F10 for Windows)
+    * Build and Reload Cmd+Shift+B (Ctrl+Shift+B for Windows)
+    * Test Package Cmd+Shift+T (Ctrl+Shift+T for Windows)
+    * Check Package Cmd+Shift+E (Ctrl+Shift+E for Windows)
+    * Document Package Cmd+Shift+D (Ctrl+Shift+D for Windows)
+<br><br>
+
+2. Push branch to github and issue a pull request (PR).
+
+3. Discuss the pull request.
+
+4. Iterate until the maintainer accepts the PR or decides it's not a good fit for VISCtemplates.
+
+This might feel overwhelming the first time you get set up, but it gets easier with practice.
+
+## Before every commit
+
+Before you commit, run the following commands one more time to make sure you didn’t break anything.
+
+* Restart R Session Cmd+Shift+F10 (Ctrl+Shift+F10 for Windows)
+* Check Package Cmd+Shift+E (Ctrl+Shift+E for Windows)
+* Document Package Cmd+Shift+D (Ctrl+Shift+D for Windows)
+
+## Pull requests
+
+Your pull request should follow these guidelines:
+
+1. **Motivation for changes.** Your pull request should clearly and concisely motivate the need for change.
+2. **Only related changes.** Before you submit your pull request, please check to make sure that you haven't accidentally included any unrelated changes. These make it harder to see exactly what's changed, and to evaluate any unexpected side effects.
+
+    Each PR corresponds to a git branch, so if you expect to submit multiple changes make sure to create multiple branches. If you have multiple changes that depend on each other, start with the first one and don't submit any others until the first one has been processed.
+
```

Comment:
What if changes have been made to the master or release branches during development of this branch? Even if someone is managing multiple changes that cascade appropriately, it doesn't stop progress by other contributors.

Reply:
Good point, and I'm currently not following this! I think this might be more important if you're contributing to a package and not part of the organization or core development team. I'm going to remove the last sentence.

Reply:
I'll normally check for changes and pull the master into my working branch -  if needed - when I am preparing to merge in my own work in. I seem to encounter this when the working branch takes time to wrap up (e.g if I am called to other priorities and a week or more has elapsed) or if the working branch depends on input from other stakeholders.